### PR TITLE
feat(contracts-bedrock): add ERC-4337 entry point v0.7.0 to predeploys

### PR DESCRIPTION
**Pull request by dancoombs**
**Description**

Adds ERC-4337 Entry Point v0.7.0 to the contract predeploy list.

**Tests**

* Simple test added to `Preinstalls.t.sol` based on previous tests. 

**Additional context**

N/A

**Metadata**

N/A

### CODE REVIEW

**Comment by dancoombs** [comment]
Associated spec PR: https://github.com/ethereum-optimism/specs/pull/228

File: op-service/predeploys/addresses.go [hunk 1]
```diff
@@ -136,12 +140,20 @@ func init() {
 		Address:       Permit2Addr,
 		ProxyDisabled: true,
 	}
-	Predeploys["SenderCreator"] = &Predeploy{
-		Address:       SenderCreatorAddr,
+	Predeploys["SenderCreator_v060"] = &Predeploy{
+		Address:       SenderCreator_v060Addr,
 		ProxyDisabled: true,
 	}
-	Predeploys["EntryPoint"] = &Predeploy{
```
Comment (dancoombs) [inline op-service/predeploys/addresses.go:143]: Want to double check that changing this name isn't backwards incompatible somewhere. Looked around the codebase and didn't find a usage, so want to check with the team.

File: packages/contracts-bedrock/src/libraries/Preinstalls.sol [hunk 2]
```diff
@@ -81,12 +87,18 @@ library Preinstalls {
...
 
+    bytes internal constant SenderCreator_v070Code =
```
Comment (dancoombs) [inline packages/contracts-bedrock/src/libraries/Preinstalls.sol:96]: https://optimistic.etherscan.io/address/0xEFC2c1444eBCC4Db75e7613d20C6a62fF67A167C

File: packages/contracts-bedrock/src/libraries/Preinstalls.sol [hunk 3]
```diff
@@ -81,12 +87,18 @@ library Preinstalls {
...
+        hex"6080600436101561000f57600080fd5b6000803560e01c63570e1a361461002557600080fd5b3461018a5760207ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffc36011261018a576004359167ffffffffffffffff9081841161018657366023850112156101865783600401358281116101825736602482870101116101825780601411610182577fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffec810192808411610155577fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe0603f81600b8501160116830190838210908211176101555792846024819482600c60209a968b9960405286845289840196603889018837830101525193013560601c5af1908051911561014d575b5073ffffffffffffffffffffffffffffffffffffffff60405191168152f35b90503861012e565b6024857f4e487b710000000000000000000000000000000000000000000000000000000081526041600452fd5b8380fd5b8280fd5b80fdfea26469706673582212207adef8895ad3393b02fab10a111d85ea80ff35366aa43995f4ea20e67f29200664736f6c63430008170033";
+
+    bytes internal constant EntryPoint_v070Code =
```
Comment (dancoombs) [inline packages/contracts-bedrock/src/libraries/Preinstalls.sol:99]: https://optimistic.etherscan.io/address/0x0000000071727De22E5E9d8BAf0edAc6f37da032

File: op-service/predeploys/addresses.go (hunk 1 above)
Comment (tynes) [reply op-service/predeploys/addresses.go:143]: Yes this is fine, I don't believe this is used anywhere anymore, previously it was used for L2 genesis generation but now that we do that in solidity, this is left over

File: packages/contracts-bedrock/src/libraries/Preinstalls.sol (hunk 3 above)
Comment (tynes) [reply packages/contracts-bedrock/src/libraries/Preinstalls.sol:99]: Confirmed that this matches

File: packages/contracts-bedrock/src/libraries/Preinstalls.sol (hunk 2 above)
Comment (tynes) [reply packages/contracts-bedrock/src/libraries/Preinstalls.sol:96]: Confirmed that this matches

**Comment by tynes** [comment]
This line in the entrypoint is pretty wild:

```
SenderCreator private immutable _senderCreator = new SenderCreator();
```

So this runs at deploy time with a create of sender creator? I don't like this syntax because its not obvious. But we put the sender creator at the same address as on op mainnet, so that means it shouldn't be a problem

File: packages/contracts-bedrock/src/libraries/Preinstalls.sol (hunk 3 above)
Comment (tynes) [reply packages/contracts-bedrock/src/libraries/Preinstalls.sol:99]: Given https://github.com/ethereum-optimism/optimism/pull/10763#issuecomment-2153887360, the sender creator address `0xEFC2c1444eBCC4Db75e7613d20C6a62fF67A167C` should be present in the bytecode of the entrypoint but I cannot find it in there. Curious what you think @dancoombs ?

Comment (dancoombs) [reply packages/contracts-bedrock/src/libraries/Preinstalls.sol:99]: Its in there in 2 places, you have to drop the `0x` and do case-insensitive search.

```
000000000000000000000000efc2c1444ebcc4db75e7613d20c6a62ff67a167c
```

**Comment by dancoombs** [comment]
> This line in the entrypoint is pretty wild:
> 
> ```
> SenderCreator private immutable _senderCreator = new SenderCreator();
> ```
> 
> So this runs at deploy time with a create of sender creator? I don't like this syntax because its not obvious. But we put the sender creator at the same address as on op mainnet, so that means it shouldn't be a problem

Yeah exactly. Easiest way to see this is to look at the entry point's creation transaction https://optimistic.etherscan.io/tx/0x1f5b834a37c7d91b9541a2b35f8d0bffcf27d4b0f2656f793478db8c8c029d6a#internal

File: packages/contracts-bedrock/src/libraries/Preinstalls.sol (hunk 3 above)
Comment (tynes) [reply packages/contracts-bedrock/src/libraries/Preinstalls.sol:99]: Ah good catch, ok thank you!

**Review by tynes — APPROVED** [review]

**Pull request merged**